### ISSUE TITLE
hl.str doesn't propagate missingness

### ISSUE THREAD

**Issue by konradjk** [hail-is/hail#4339]
```
x = hl.utils.range_table(10).annotate(x=hl.or_missing(hl.rand_bool(0.5), hl.rand_bool(0.5)))
x = x.annotate(y=hl.str(x.x))
x = x.annotate(z=x.y == "NA")
x.show()
+-------+-------+-------+-------+
|   idx | x     | y     | z     |
+-------+-------+-------+-------+
| int32 | bool  | str   | bool  |
+-------+-------+-------+-------+
|     0 | false | false | false |
|     1 | NA    | NA    | true  |
|     2 | true  | true  | false |
|     3 | NA    | NA    | true  |
|     4 | NA    | NA    | true  |
|     5 | true  | true  | false |
|     6 | NA    | NA    | true  |
|     7 | true  | true  | false |
|     8 | true  | true  | false |
|     9 | true  | true  | false |
+-------+-------+-------+-------+
```

**Comment by tpoterba** [comment]
fixed by #4348

**Comment by chrisvittal** [comment]
#4348 in master. closing
